packages/meditrak-server/src/routes/GETAccessRequests/assertAccessRequestPermissions.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import {
+  BES_ADMIN_PERMISSION_GROUP,
+  TUPAIA_ADMIN_PANEL_PERMISSION_GROUP,
+} from '../../permissions/constants';
+
+export const hasAccessRequestPermissions = async (accessPolicy, models, accessRequest) => {
+  const entity = await models.entity.findById(accessRequest.entity_id);
+  return accessPolicy.allows(entity.country_code, TUPAIA_ADMIN_PANEL_PERMISSION_GROUP);
+};
+
+export const assertAccessRequestPermissions = async (accessPolicy, models, accessRequest) => {
+  const hasPermission = await hasAccessRequestPermissions(accessPolicy, models, accessRequest);
+  if (!hasPermission) {
+    throw new Error('Need Admin Panel access to the country this access request is for');
+  }
```

Comment:
To avoid duplicate code of checking accessPolicy, I think you can change this to:
```suggestion
  const results = await filterAccessRequestsByPermissions (accessPolicy, [accessRequest], models);
  if (results.length === 0) {
    throw new Error('Need Admin Panel access to the country this access request is for');
  }
```

Then you can remove the method `hasAccessRequestPermissions`.

Same goes to other assertPermissions logic

Reply:
Good call. I think I started this pattern when intending to use the `hasAccess` function as the array filter, and have just kept doing it out of habit.